File: calendarbackend/controllers/eventsController.js
const events = require("../models/events");

module.exports = {


  index(req, res) {
    console.log("getting index")
    events.findAll()
     .then(events => {
       res.json({
        message: 'ok',
        data: { events },
       });
      })
      .catch(err => {
        console.log(err);
        res.status(400).json({message: '400', err});
      });
    },


    getDay(req, res) {
    events.findById(req.params.day_id)
    .then(event => {
      res.json({
        message: 'ok',
        data: { event },
      });
    })
    .catch(err => {
        console.log(err);
        res.status(400).json({message: '400', err});
      });
    },

    create(req, res) {
    events.save({
      start_time: req.body.start_time,
      end_time: req.body.end_time,
      description: req.body.description,
      day_id: req.body.day_id
    })
    .then(event => {
      res.json({message: 'ok', data: { event }});
    })
    .catch(err => {
      console.log(err);
      res.status(400).json({message: '400', err});
    });
  },


   destroy(id) {
    return db.none(`
    DELETE
    FROM events
    WHERE id = $1
    `, id);
  },


};

File: calendarbackend/app/controllers/events_controller.rb
class EventsController < ApplicationController

	def index
		#@events = Event.all


		#if stale?(last_modified: @events.updated_at, public: true)
			#memecache to check browser cache
		@events = Event.order('day_id', 'start_time')
		#render json: {status: 'SUCCESS', message: 'Loaded Events', data:@events}, status: :ok
		#render json: @events, status: :ok
		render json: @events 
	#end
		
	end

	def show
		@events = Event.where("day_id = ?", params[:id]).order('start_time')
		

		#if stale?(last_modified: @events.updated_at, public: true)
		render json: @events
	#end
	#end
	end


	def create
		@event = Event.new(event_params)
		#@event = Event.new(params.require(:event).permit(:start_time, :end_time, :description, :day_id))
		if @event.save
			@events = Event.where("day_id = ?", params[:day_id]).order('start_time')
			#@events = Event.all
			render json: @events
		else
			head :unprocessable_entity
		end
	end


	def update
		@event = Event.find(params[:id])
		if @event.update(event_params)
			#@events = Event.all
			@events = Event.where("day_id = ?", params[:day_id]).order('start_time')
			render json: @events
		else
			head :unprocessable_entity
		end

	end

	def destroy
		@event = Event.find(params[:id])
		if @event.destroy
			render json: @event
		else
			head::unprocessable_entity
		end
	end



	private
	def event_params
		params.permit(:start_time, :end_time, :description, :day_id)
	end




end
